src/Compilers/CSharp/Test/Semantic/Semantics/SimpleLambdaParametersWithModifiersTests.cs

```diff
@@ -0,0 +1,1458 @@
+                Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfTarget, "(a, ref b) => { }").WithArguments("a", "D").WithLocation(8, 15));
+
+        var tree = compilation.SyntaxTrees.Single();
+        var root = tree.GetRoot();
+        var lambda = root.DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+
+        var semanticModel = compilation.GetSemanticModel(tree);
+        var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol!;
+
+        Assert.Equal(MethodKind.LambdaMethod, symbol.MethodKind);
+        Assert.Equal(ScopedKind.None, symbol.Parameters.First().ScopedKind);
+        Assert.Equal(compilation.GetTypeByMetadataName(typeof(ReadOnlySpan<>).FullName).GetPublicSymbol(), symbol.Parameters.First().Type.OriginalDefinition);
+    }
+
+    [Fact]
+    public void TestOneParameterWithScopedAndOptionalValue()
+    {
+        var compilation = CreateCompilationWithSpan("""
+            using System;
+            delegate void D(scoped ReadOnlySpan<int> x);
+
+            class C
+            {
+                void M()
+                {
+                    D d = (scoped x = default) => { };
+                }
+            }
+            """).VerifyDiagnostics(
+                // (8,23): error CS9098: Implicitly typed lambda parameter 'x' cannot have a default value.
+                //         D d = (scoped x = default) => { };
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedDefaultParameter, "x").WithArguments("x").WithLocation(8, 23),
+                // (8,23): warning CS9099: Parameter 1 has default value 'null' in lambda but '<missing>' in the target delegate type.
+                //         D d = (scoped x = default) => { };
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "x").WithArguments("1", "null", "<missing>").WithLocation(8, 23));
+
+        var tree = compilation.SyntaxTrees.Single();
+        var root = tree.GetRoot();
+        var lambda = root.DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+
+        var semanticModel = compilation.GetSemanticModel(tree);
+        var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol!;
+
+        Assert.Equal(MethodKind.LambdaMethod, symbol.MethodKind);
+        Assert.Equal(ScopedKind.ScopedValue, symbol.Parameters.Single().ScopedKind);
+        Assert.True(symbol.Parameters.Single().IsOptional);
+        Assert.Equal(compilation.GetTypeByMetadataName(typeof(ReadOnlySpan<>).FullName).GetPublicSymbol(), symbol.Parameters.Single().Type.OriginalDefinition);
+    }
+
+    [Theory, CombinatorialData]
+    public void TestOneParameterWithScopedAsParameterName(bool escaped)
+    {
+        var compilation = CreateCompilationWithSpan($$"""
+            using System;
+            delegate void D(scoped ReadOnlySpan<int> x);
+
+            class C
+            {
+                void M()
+                {
+                    D d = (scoped {{(escaped ? "@" : "")}}scoped) => { };
```

Comment:
>(scoped {{(escaped ? "@" : "")}}scoped) => { } [](http://example.com/codeflow?start=26&length=46)

I'm surprised the second `scoped` is treated as a modifier in `(scoped scoped x) => { }` but as an identifier in `(scoped scoped) => { }`.

Reply:
It's less about this being a context (like query-expressions) where a word just unilaterally becomes a keyword.  Instead, it's more like: in the case where it could multiple things (like a modifier vs the type), then it's only a modifier (even though that breaks from prior interpretations).

The 'identifier' part is always mandatory, and never ambiguous. So it's just an identifier there.